kedro_azureml/datasets.py

```diff
@@ -85,3 +93,84 @@ def _save(self, data: Any) -> None:
             logger.warning(
                 f"DataSet {self.dataset_name} will not be saved on a distributed node"
             )
+
+
+# TODO: Switch to use Azure ML File dataset?
+class AzureMLFolderDataset(AbstractDataSet):
+    def __init__(
+        self,
+        path: str,
+        dataset: Union[str, Type[AbstractDataSet], Dict[str, Any]],
+        filepath_arg: str = "filepath",
+    ):
+        """Creates a new instance of ``AzureMLFolderDataset``.
+
+        Args:
+            path: Path to the file containing the data.
+            dataset: Underlying dataset definition.
+                Accepted formats are:
+                a) object of a class that inherits from ``AbstractDataSet``
+                b) a string representing a fully qualified class name to such class
+                c) a dictionary with ``type`` key pointing to a string from b),
+                other keys are passed to the Dataset initializer.
+            filepath_arg: Underlying dataset initializer argument that will
+                contain a path to each corresponding partition file.
+                If unspecified, defaults to "filepath".
+
+        Raises:
+            DataSetError: If versioning is enabled for the underlying dataset.
+        """
+
+        super().__init__()
+
+        self.path = path
+
+        dataset = dataset if isinstance(dataset, dict) else {"type": dataset}
+        self._dataset_type, self._dataset_config = parse_dataset_definition(dataset)
+
+        # TODO: remove and disable versioning in Azure ML runner?
```

Comment:
Any opinion here @marrrcin? I think it would be nice to support versioning of the underlying dataset when running the pipeline locally. We could dynamically disable the versioning then in the Azure ML runner.